src/components/new-suggestion/areas/review.area.tsx

```diff
@@ -1,16 +1,162 @@
-import React from "react"
-import { useSelector } from "react-redux"
+import React, { useState } from "react"
+import { useDispatch, useSelector } from "react-redux"
 import { AppState } from "@/redux/store"
+import Spinner from "@/components/utils/spinner"
+import { MusicalNoteIcon } from "@heroicons/react/24/solid"
+import { ReviewPlaceholder } from "@/constants/review-placeholder"
+import { insertSuggestion, insertSuggestionInstruments } from "@/services/suggestion.service"
+import ErrorPopup from "@/components/popups/error-popup"
+import { SuggestionInstrumentDatabaseOperation } from "@/types/database-types"
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react"
+import { Database } from "@/types/database"
+import { useRouter } from "next/router"
+import {
+  initialState,
+  setActiveArea,
+  updateNewSuggestion,
+} from "@/redux/slices/new-suggestion.slice"
+import { Area } from "@/constants/area"
+import { NewSuggestionInstrument } from "@/interfaces/new-suggestion"
+import { getInstrumentImage } from "@/helpers/cloudinary.helper"
+import Image from "next/image"
 
 const ReviewArea = () => {
-  const newSuggestion = useSelector((state: AppState) => state.newSuggestion.suggestion)
+  const supabase = useSupabaseClient<Database>()
+  const suggestion = useSelector((state: AppState) => state.newSuggestion.suggestion)
+  const [showSpinner, setShowSpinner] = useState<boolean>(false)
+  const [insertError, setInsertError] = useState<boolean>(false)
+  const dispatch = useDispatch()
+  const router = useRouter()
+  const user = useUser()
+
+  const saveSuggestion = () => {
+    setShowSpinner(true)
+    insertSuggestion(supabase, suggestion, user!.id)
+      .then((response) => {
+        if (response.error) {
+          handleError()
+          return
+        }
+
+        const suggestionId = response.data.at(0)!.id
+        insertSuggestionInstruments(supabase, mapInstruments(suggestionId))
+          .then((response) => {
+            if (response.error) {
+              handleError()
+              return
+            }
+
+            dispatch(updateNewSuggestion(initialState.suggestion))
+            dispatch(setActiveArea(Area.SongInformation))
+            router.push("/suggestions")
+          })
+          .catch(() => handleError())
+      })
+      .catch(() => handleError())
+      .finally(() => {
+        setShowSpinner(false)
```

Comment:
https://github.com/ASVGay/the-rhapsodies/assets/54884540/660bdbb6-dd2c-4e9e-a5bc-02319e37eddc

The song information area is shown for a bit (probably longer than a bit with slower connections) after submitting the suggestion. I think this is because setShowSpinner is set to false before router.push on line 51 is resolved